tools/testsys/src/aws_resources.rs

```diff
@@ -49,13 +59,74 @@ impl AwsK8s {
         sonobuoy_mode: SonobuoyMode,
     ) -> Result<Vec<Crd>> {
         let crds = vec![
-            self.eks_crd("", testsys_images)?,
-            self.ec2_crd("", testsys_images)?,
-            self.sonobuoy_crd("", "-test", sonobuoy_mode, None, testsys_images)?,
+            self.eks_crd(testsys_images)?,
+            self.ec2_crd(testsys_images, None)?,
+            self.sonobuoy_crd("-test", sonobuoy_mode, None, testsys_images)?,
         ];
         Ok(crds)
     }
 
+    /// Creates 5 `Test` crds for migration testing. Consisting of a `quick` test, a migration to
+    /// the target version, a `quick` test to validate the migration, a migration to the starting
+    /// version, and a `quick` test to validate the second migration.
+    async fn migration_test_crds(&self, testsys_images: &TestsysImages) -> Result<Vec<Crd>> {
+        let ami = self
+            .starting_image_id
+            .as_ref()
+            .unwrap_or(
+                &get_ami_id(
+                    format!(
+                        "bottlerocket-{}-{}-{}-{}",
+                        self.variant, self.arch, self.starting_version.as_ref().context("The starting version must be provided for migration testing")?, self.migrate_starting_commit.as_ref().context("The commit for the starting version must be provided if the starting image id is not")?
+                    ), & self.arch,
+                    self.region.to_string(),
+                )
+                .await?,
+            )
+            .to_string();
+        let eks = self.eks_crd(testsys_images)?;
+        let ec2 = self.ec2_crd(testsys_images, Some(ami))?;
+        let mut depends_on = Vec::new();
+        let initial = self.sonobuoy_crd("-1-initial", SonobuoyMode::Quick, None, testsys_images)?;
+        depends_on.push(initial.name().context("Crd missing name")?);
+        let start_migrate = self.migration_crd(
+            format!("{}-2-migrate", self.cluster_name()),
+            MigrationVersion::Migrated,
+            Some(depends_on.clone()),
+            testsys_images,
+        )?;
+        depends_on.push(start_migrate.name().context("Crd missing name")?);
```

Comment:
Also not a big deal here but should "name" be a required field or have a default value?

I wonder if the cases where the name is optional are worth the added friction in what feels like the normal case here. You could potentially have a `NamedCrd` type and add a `From` impl to `Crd` for it.

Reply:
The crd itself (for tests and resources) has the name field as an `Option` so somewhere there needs to be an `unwrap()` or `context()` to read it unless we default with  `""`, but that has the potential to cause issues if the field was missing for some reason.